_overviews/tutorials/binary-compatibility-for-library-authors.md

```diff
@@ -0,0 +1,215 @@
+
+How can we, as library authors, spare our users of runtime errors and dependency hell?
+
+* Use **Migration Manager** (MiMa) to catch unintended binary compatibility breakages before releasing a new library version
+* **Avoid breaking binary compatibility** through careful design and evolution of your library interfaces
+* Communicate binary compatibility breakages clearly through **versioning**
+
+## MiMa - Checking binary compatibility against previous library versions 
+
+The [Migration Manager for Scala](https://github.com/typesafehub/migration-manager) (MiMa) is a tool for diagnosing binary incompatibilities between different library versions.  
+It works by comparing the class files of two provided JARs and report any binary incompatibilities found.
+
+By incorporating [MiMa SBT plugin](https://github.com/typesafehub/migration-manager/wiki/Sbt-plugin) into your SBT build, you can easily check whether 
+you have accidentally introduced binary incompatible changes. Detailed instruction on how to use the SBT plugin can be found in the link.
+
+We strongly encourage every library author to incorporate MiMa into their library release workflow.
+
+## Evolving code without breaking binary compatibility
+
+Binary compatibility breakages can often be avoided through careful use of certain Scala features as well as some techniques you can apply when modifying code.
+
+Some language features may break binary compatibility:
+
+* Default parameter values for methods or classes
+* Case classes
+* Default methods on traits (doesn't cause breakages since 2.12)
+
+Techniques you can use to avoid breaking binary compatibility:
+
+* Annotate public method's return type explicitly
+* Mark methods as package private when you want to remove a method or modify its signature
+* Don't turn on inlining for methods from dependencies
+
+For brevity of this guide, detailed explanation and runnable code examples can be found in [Binary Compatibility Code Examples & Explanation](https://github.com/jatcwang/binary-compatibility-guide).
+
+Again, we recommend using MiMa to double check that you have not broken binary compatibility after making changes.
+
+## Versioning Scheme - Communicating compatibility breakages
+
+Library authors use versioning schemes to communicate compatibility guarantees between library releases to their users. Versioning schemes like [Semantic Versioning](http://semver.org/)(SemVer) allow
+users to easily reason about the impact of updating a library, without needing to read the detailed release note.
+
+In the following section, we will outline a versioning scheme based on Semantic Versioning that we **strongly encourage** you to adopt for your libraries. The rules listed below are **in addition** to 
+Semantic Versioning v2.0.0.
+
+### Recommended Versioning Scheme
+
+* If backward **binary compatibility** is broken, **major version number** must be increased
```

Comment:
You miss a '.' at the end of every item in the enumeration. I think it's a convention to put either ';` or `.` instead of not ending the sentence with any punctuation mark.

Reply:
In English typography, there are no hard rules on the proper use of bulleted lists.

The only well-accepted rule I have seen is that, if the full list can read as a full sentence, you can write the list as if bullets were basically not semantically relevant. Therefore, the rest of the punctuation and capitalization should be applied as if they were all part of the same sentence. This means that a bullet list should look as follows:
* items should not start with a capital letter,
* they should all end with a comma, and
* the next-to-last item ends with ", and" while the last one ends with a period.

Indeed, you can remove the bullets from the above list, and it still reads as a correct sentence.

Reply:
Cool, thanks for letting me know! Myself I've always used `;` and then, for the last item in an enumeration, `; and,`. I'll use comma from now on.

Reply:
I'm gonna use full stops as they are all full sentences